KubuQA.sh

```diff
@@ -78,46 +101,28 @@ check_and_install_tool zsync zsync
 check_and_install_tool wget wget
 check_and_install_tool VBoxManage virtualbox
 
-# Check if an TestKubuntuInstall VM exists
+# Check whether various components exist. If not or if requested, (re)create them
 check_existing_vm
-
-# Ensure the ISO Download directory exists
-mkdir -p "$directory"
-cd "$directory"
-
-# Check if the ISO file exists, and has already been downloaded
-if [ -f "$isoFileName" ]; then
-    # Prompt the user to check for updates
-    if kdialog --yesno "I found an ISO Test Image, would you like to check for updates?"; then
-        # Use zsync to update the ISO
-        zsync "http://$isoDownloadURL.zsync"
-    fi
-else
-    # Prompt the user to download the ISO if it doesn't exist
-    if kdialog --yesno "No local test ISO image available, should I download one?"; then
-        # Download the ISO
-        wget "https://$isoDownloadURL"
-    else
-        exit
-    fi
-fi
+check_existing_vdi
+check_existing_iso
 
 # Prompt the user to launch a test install using VirtualBox
 if kdialog --yesno "Launch a Test Install using Virtual Box?"; then
-    # Use VirtualBox to launch a VM booting from the ISO image
-    VBoxManage createvm --name "TestKubuntuInstall" --register
 
     # Enable the user to choose which device to boot from
     choice=$(kdialog --menu "Select boot medium" 1 "ISO" 2 "HDD")
 
     case "$choice" in
-        1) VBoxManage modifyvm "TestKubuntuInstall" --memory 2048 --acpi on --boot1 dvd --nic1 nat ;;
+           # Connect the ISO to its storage controller and make VirtualBox boot from it
+        1) VBoxManage modifyvm "TestKubuntuInstall" --memory 2048 --acpi on --boot1 dvd --nic1 nat
+           VBoxManage storageattach "TestKubuntuInstall" --storagectl "IDE Controller" --port 0 --device 0 --type dvddrive --medium "$isoFilePath" ;;
+           # Make VirtualBox boot from the VDI
         2) VBoxManage modifyvm "TestKubuntuInstall" --memory 2048 --acpi on --boot1 disk --nic1 nat ;;
         *) echo "Invalid choice"; exit 1 ;;
     esac
 
-    VBoxManage storagectl "TestKubuntuInstall" --name "IDE Controller" --add ide
-    VBoxManage storageattach "TestKubuntuInstall" --storagectl "IDE Controller" --port 0 --device 0 --type hdd --medium "$vdi_file"
+    # Connect the VDI to its storage controller
+    VBoxManage storageattach "TestKubuntuInstall" --storagectl "SATA Controller" --port 0 --device 0 --type hdd --medium "$vdi_file"
```

Comment:
I looked at a manually created Kubtuntu VM and noticed that the ISO is connected via a IDE controller, and the VDI via a SATA controller. I don't know why and what the difference between those are, but since VirtualBox does it like this out of the box, I guess we should as well.